v2/yarpcrouter/router.go

```diff
@@ -58,19 +59,78 @@ type MapRouter struct {
 func NewMapRouter(defaultService string) MapRouter {
 	return MapRouter{
 		defaultService:            defaultService,
-		serviceProcedureEncodings: make(map[serviceProcedureEncoding]yarpc.Procedure),
+		serviceProcedureEncodings: make(map[serviceProcedureEncoding]yarpc.TransportProcedure),
 		serviceNames:              map[string]struct{}{defaultService: {}},
 	}
 }
 
+// MapToTransportProcedures converts encoding-level procedures to transport-level procedures
+func MapToTransportProcedures(encodingProcedures []yarpc.EncodingProcedure) []yarpc.TransportProcedure {
+	transportProcedures := []yarpc.TransportProcedure{}
```

Comment:
It looks like there's a 1:1 correspondence with our encoding procedures and transport procedures. We can allocate the length ahead of time:
```
transportProcedures := make([]yarpc.TransportProcedure, len(encodingProcedures))
for i, p := range encodingProcedures {
  ...
  transportProcedures[i] = yarpc.TransportProcedure{ ... }
}
```